sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryExpression.cs

```diff
@@ -14,21 +14,22 @@ namespace Azure.Core.Expressions.DataFactory
     /// </summary>
     /// <typeparam name="T"> Can be one of <see cref="string"/>, <see cref="bool"/>, <see cref="int"/>, <see cref="double"/>, <see cref="Array"/>. </typeparam>
 #pragma warning disable SA1649 // File name should match first type name
-    public sealed partial class DataFactoryExpression<T> : IUtf8JsonSerializable
+    [JsonConverter(typeof(DataFactoryExpressionJsonConverter))]
+    public sealed class DataFactoryExpression<T>
 #pragma warning restore SA1649 // File name should match first type name
     {
-        private string? _type;
-        private T? _literal;
-        private string? _expression;
+        internal string? Type { get; }
+        internal T? LiteralInternal { get; }
+        internal string? Expression { get; }
 
         /// <summary>
         /// Initializes a new instance of DataFactoryExpression with a literal value.
         /// </summary>
-        /// <param name="literal"> The literal value. </param>
-        public DataFactoryExpression(T literal)
+        /// <param name="literalInternal"> The literal value. </param>
+        public DataFactoryExpression(T literalInternal)
```

Comment:
not sure we want this rename on this public parameter, lets keep it `literal`.

Reply:
woops this was unintended